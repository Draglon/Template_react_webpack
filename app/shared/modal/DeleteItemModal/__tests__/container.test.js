import React from 'react';
import configureStore from 'redux-mock-store';
import { Modal } from 'antd';

import DeleteItemModalContainer from '../container';

const spy = jest.spyOn(Modal, 'confirm');

describe('<DeleteItemModalContainer />', () => {
  const props = {
    title: 'some title',
    params: { movieId: 9999 },
    onConfirm: jest.fn(),
  };

  const store = configureStore()({});
  const container = shallow(<DeleteItemModalContainer store={store} {...props} />);
  const instance = container.instance();

  describe('showModal()', () => {
    it('should call showModal()', () => {
      instance.showModal();

      expect(spy).toHaveBeenCalledWith({
        title: props.title,
        onOk: expect.any(Function),
      });
    });
  });

  describe('onConfirm()', () => {
    it('should call onConfirm()', () => {
      instance.onOk();

      expect(props.onConfirm).toHaveBeenCalledWith(props.params);
    });
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
