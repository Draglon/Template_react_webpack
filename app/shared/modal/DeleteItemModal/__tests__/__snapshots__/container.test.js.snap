// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteItemModalContainer /> should match snapshot 1`] = `
<DeleteItemModalComponent
  onConfirm={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "movieId": 9999,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  params={
    Object {
      "movieId": 9999,
    }
  }
  showModal={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  title="some title"
/>
`;
